tests/test-update-install-flatpaks.c

```diff
@@ -3906,6 +3906,338 @@ test_update_uninstall_flatpaks_on_reboot (EosUpdaterFixture *fixture,
+                                                                 NULL,
+                                                                 "test-repo",
+                                                                 TRUE,
+                                                                 "app/org.test.Test/arch/stable",
+                                                                 NULL));
+
+  /* Set up repos */
+  g_hash_table_insert (flatpak_repo_infos,
+                       g_strdup ("test-repo"),
+                       flatpak_repo_info_new ("test-repo",
+                                              "com.endlessm.TestInstallFlatpaksCollection",
+                                              "com.endlessm.TestInstallFlatpaksCollection"));
+
+  eos_test_setup_flatpak_repo (updater_directory,
+                               flatpak_install_infos,
+                               flatpak_repo_infos,
+                               gpg_key_file,
+                               keyid,
+                               &error);
+
+  /* Update the server, so it has a new commit (1).
+   */
+  etc_update_server (data, 1);
+  /* Update the client, so it also has a new commit (1); and, at this
+   * point, two deployments - old one pointing to commit 0 and a new
+   * one pointing to commit 1.
+   */
+  etc_update_client (data);
+
+  /* Now simulate a reboot by running eos-updater-flatpak-installer */
+  deployment_csum = get_checksum_for_deploy_repo_dir (deployment_repo_dir,
+                                                      refspec,
+                                                      &error);
+  g_assert_no_error (error);
+
+  eos_test_run_flatpak_installer (data->client->root,
+                                  deployment_csum,
+                                  default_remote_name,
+                                  &error);
+  g_assert_no_error (error);
+
+  /* Update the server again with commit 2, then update the client. This will
+   * pull in a new ref-action list with the uninstall action. */
+  etc_update_server (data, 2);
+  etc_update_client (data);
+
+  /* Reboot again, run the flatapk installer over this new commit */
+  deployment_csum = get_checksum_for_deploy_repo_dir (deployment_repo_dir,
+                                                      refspec,
+                                                      &error);
+  g_assert_no_error (error);
+
+  eos_test_run_flatpak_installer (data->client->root,
+                                  deployment_csum,
+                                  default_remote_name,
+                                  &error);
+  g_assert_no_error (error);
+
+  /* Extension should not be installed here */
+  deployed_flatpaks = eos_test_get_installed_flatpaks (updater_directory, &error);
+  g_assert_no_error (error);
+
+  g_assert_true (g_strv_contains ((const gchar * const *) deployed_flatpaks,
+                                  "org.test.Test.Extension"));
```

Comment:
You should also verify in the middle step that both the app and extension are installed; and you should verify here that the app is uninstalled.

Reply:
Added these for now

Reply:
I’m not seeing them, but I could be being confused by diff helpfully adding the new tests as modifications of the old ones. Did you definitely push the change which added them?